### DISCUSSION TITLE
v4.1.0a2

### DISCUSSION THREAD

**Discussion by hardbyte** [hardbyte/python-can#1441] [Announcements]
We're about ready to publish version `4.1.0` and would really appreciate a few people installing the pre-release from pypi and commenting here that it worked (and on what operating system + interface) - or opening an issue so we can address anything before releasing the final version.

To install the pre-release use:

```
pip install python-can==4.1.0a2
```

<hr /><em>This discussion was created from the release <a href='https://github.com/hardbyte/python-can/releases/tag/v4.1.0a2'>v4.1.0a2</a>.</em>

**Comment by hardbyte** [comment]
I installed the pre-release on Python3.10 on linux and ran a few of the examples against the socketcan interface using a virtual device. ✅ 

<details>
    <summary>console</summary>
<pre>
┌─[brian@hardbyte-nzxt] - [~/tmp] - [Fri Nov 18, 19:46]
└─[$] <> mkdir test-python-can-4.1.0a2
┌─[brian@hardbyte-nzxt] - [~/tmp] - [Fri Nov 18, 19:46]
└─[$] <> cd test-python-can-4.1.0a2 
┌─[brian@hardbyte-nzxt] - [~/tmp/test-python-can-4.1.0a2] - [Fri Nov 18, 19:46]
└─[$] <> python3.10 -m venv venv
┌─[brian@hardbyte-nzxt] - [~/tmp/test-python-can-4.1.0a2] - [Fri Nov 18, 19:46]
└─[$] <> source ./venv/bin/activate
┌─[brian@hardbyte-nzxt] - [~/tmp/test-python-can-4.1.0a2] - [Fri Nov 18, 19:46]
└─[$] <> pip install python-can==4.1.0a2
Collecting python-can==4.1.0a2
  Downloading python_can-4.1.0a2-py3-none-any.whl (241 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 242.0/242.0 kB 2.8 MB/s eta 0:00:00
Collecting msgpack~=1.0.0
  Using cached msgpack-1.0.4-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (316 kB)
Requirement already satisfied: setuptools in ./venv/lib/python3.10/site-packages (from python-can==4.1.0a2) (63.2.0)
Collecting wrapt~=1.10
  Using cached wrapt-1.14.1-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (77 kB)
Collecting packaging
  Using cached packaging-21.3-py3-none-any.whl (40 kB)
Collecting typing-extensions>=3.10.0.0
  Using cached typing_extensions-4.4.0-py3-none-any.whl (26 kB)
Collecting pyparsing!=3.0.5,>=2.0.2
  Using cached pyparsing-3.0.9-py3-none-any.whl (98 kB)
Installing collected packages: msgpack, wrapt, typing-extensions, pyparsing, packaging, python-can
Successfully installed msgpack-1.0.4 packaging-21.3 pyparsing-3.0.9 python-can-4.1.0a2 typing-extensions-4.4.0 wrapt-1.14.1

[notice] A new release of pip available: 22.2.2 -> 22.3.1
[notice] To update, run: pip install --upgrade pip
┌─[brian@hardbyte-nzxt] - [~/tmp/test-python-can-4.1.0a2] - [Fri Nov 18, 19:46]
└─[$] <> python -m can.logger -i socketcan -c vcan0
Connected to SocketcanBus: socketcan channel 'vcan0'
Can Logger (Started on 2022-11-18 19:48:00.804194)
Timestamp: 1668754085.671626        ID: 0401    S Tx                DL:  6    11 11 11 11 11 11           Channel: vcan0
Timestamp: 1668754086.671639        ID: 0401    S Tx                DL:  6    22 22 22 22 22 22           Channel: vcan0
Timestamp: 1668754087.671644        ID: 0401    S Tx                DL:  6    33 33 33 33 33 33           '333333'    Channel: vcan0
Timestamp: 1668754088.671655        ID: 0401    S Tx                DL:  6    44 44 44 44 44 44           'DDDDDD'    Channel: vcan0
Timestamp: 1668754089.671660        ID: 0401    S Tx                DL:  6    55 55 55 55 55 55           'UUUUUU'    Channel: vcan0
Timestamp: 1668754090.671668        ID: 0401    S Tx                DL:  6    11 11 11 11 11 11           Channel: vcan0
Timestamp: 1668754091.671684        ID: 0401    S Tx                DL:  6    22 22 22 22 22 22           Channel: vcan0
Timestamp: 1668754092.671703        ID: 0401    S Tx                DL:  6    33 33 33 33 33 33           '333333'    Channel: vcan0
Timestamp: 1668754093.671724        ID: 0401    S Tx                DL:  6    44 44 44 44 44 44           'DDDDDD'    Channel: vcan0
Timestamp: 1668754094.671729        ID: 0401    S Tx                DL:  6    55 55 55 55 55 55           'UUUUUU'    Channel: vcan0
Timestamp: 1668754095.671735        ID: 0401    S Tx                DL:  6    11 11 11 11 11 11           Channel: vcan0
Timestamp: 1668754095.681483        ID: 0401    S Tx                DL:  6    11 11 11 11 11 11           Channel: vcan0
Timestamp: 1668754096.681494        ID: 0401    S Tx                DL:  6    33 33 33 33 33 33           '333333'    Channel: vcan0
Timestamp: 1668754097.681499        ID: 0401    S Tx                DL:  6    55 55 55 55 55 55           'UUUUUU'    Channel: vcan0
Timestamp: 1668754098.681503        ID: 0401    S Tx                DL:  6    11 11 11 11 11 11           Channel: vcan0
Timestamp: 1668754099.681508        ID: 0401    S Tx                DL:  6    33 33 33 33 33 33           '333333'    Channel: vcan0
Timestamp: 1668754100.681514        ID: 0401    S Tx                DL:  6    55 55 55 55 55 55           'UUUUUU'    Channel: vcan0
Timestamp: 1668754101.681519        ID: 0401    S Tx                DL:  6    11 11 11 11 11 11           Channel: vcan0
Timestamp: 1668754102.681524        ID: 0401    S Tx                DL:  6    33 33 33 33 33 33           '333333'    Channel: vcan0
Timestamp: 1668754103.681649        ID: 0401    S Tx                DL:  6    55 55 55 55 55 55           'UUUUUU'    Channel: vcan0
Timestamp: 1668754104.681748        ID: 0401    S Tx                DL:  6    22 22 22 22 22 22           Channel: vcan0
Timestamp: 1668754105.681770        ID: 0401    S Tx                DL:  6    44 44 44 44 44 44           'DDDDDD'    Channel: vcan0
Timestamp: 1668754106.681865        ID: 0401    S Tx                DL:  6    66 66 66 66 66 66           'ffffff'    Channel: vcan0
Timestamp: 1668754107.681906        ID: 0401    S Tx                DL:  6    22 22 22 22 22 22           Channel: vcan0
Timestamp: 1668754108.681922        ID: 0401    S Tx                DL:  6    44 44 44 44 44 44           'DDDDDD'    Channel: vcan0
Timestamp: 1668754109.681967        ID: 0401    S Tx                DL:  6    66 66 66 66 66 66           'ffffff'    Channel: vcan0
Timestamp: 1668754110.681972        ID: 0401    S Tx                DL:  6    22 22 22 22 22 22           Channel: vcan0
Timestamp: 1668754111.682019        ID: 0401    S Tx                DL:  6    44 44 44 44 44 44           'DDDDDD'    Channel: vcan0
Timestamp: 1668754112.682126        ID: 0401    S Tx                DL:  6    66 66 66 66 66 66           'ffffff'    Channel: vcan0
Timestamp: 1668754113.682132        ID: 0401    S Tx                DL:  6    22 22 22 22 22 22           Channel: vcan0
</pre>
</details>

**Comment by zariiii9003** [comment]
I'm using 4.1.0a0 with Windows & Vector since release. No issues here.
